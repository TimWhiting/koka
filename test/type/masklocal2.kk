// emit a string
effect fun emit( s : string ) : ()

fun pretty-emit1(action : () -> <console,emit,exn|e> a) : <console|e> error<a>
  try
    with fun emit(s) s.println
    emit("hi")
    action()

fun pretty-emit2(action : () -> <emit|e> a) : e (a,string)
  var out := ""
  with fun emit(s) out := out ++ s ++ "\n"
  (action(),out)

fun pretty-emit3(action : () -> <emit|e> a) : e string
  var out := ""
  with fun emit(s) out := out ++ s ++ "\n"
  action()
  out
