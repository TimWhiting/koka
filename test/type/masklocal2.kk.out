type/masklocal2/emit: (s : string) -> emit ()
type/masklocal2/pretty-emit1: forall<a,e> (action : () -> <console,emit,exn|e> a) -> <console|e> error<a>
type/masklocal2/pretty-emit2: forall<a,e> (action : () -> <emit|e> a) -> e (a, string)
type/masklocal2/pretty-emit3: forall<a,e> (action : () -> <emit|e> a) -> e string